test/compilable/vcg-ast-arraylength.d

```diff
@@ -12,4 +12,12 @@ void main()
 
 	// should convert to arr = arr[0 .. 0];
 	arr.length = 0;
+
+    // https://issues.dlang.org/show_bug.cgi?id=21678
+    int[] f;
+    int[] a;
+
+    a.length = f.length = 0;
+    const x = 0;
+    a.length = f.length = x;
```

Comment:
Add a check to verify that the type of the expression is size_t, e.g.

`static assert(is(typeof(a.length = 0) == size_t)));`

Reply:
Test added.
I had to force the type to the same as the length, good catch.